src/main/java/io/neonbee/internal/handler/DefaultErrorHandler.java

```diff
@@ -77,6 +76,71 @@ public void handle(RoutingContext routingContext) {
         answerWithError(routingContext, errorCode, errorMessage);
     }
 
+    /**
+     * Creates a String with the text/plain error response.
+     *
+     * @param routingContext the routing context
+     * @param errorCode      the error code of the response
+     * @param errorMessage   the error message of the response
+     * @return A String with the plain text error response.
+     */
+    protected String createPlainResponse(RoutingContext routingContext, int errorCode, String errorMessage) {
+        StringBuilder builder =
+                new StringBuilder().append("Error ").append(errorCode).append(": ").append(errorMessage);
+        String correlationId = getCorrelationId(routingContext);
+        if (correlationId != null) {
+            builder.append(" (Correlation ID: ").append(correlationId).append(')');
+        }
+        return builder.toString();
+    }
+
+    /**
+     * Creates a String with the application/json error response.
+     *
+     * @param routingContext the routing context
+     * @param errorCode      the error code of the response
+     * @param errorMessage   the error message of the response
+     * @return A String with the error response in JSON format.
+     */
+    protected String createJsonResponse(RoutingContext routingContext, int errorCode, String errorMessage) {
+        JsonObject jsonResponse = new JsonObject().put("code", errorCode).put("message", errorMessage);
+        String correlationId = getCorrelationId(routingContext);
+        if (correlationId != null) {
+            jsonResponse.put("correlationId", correlationId);
+        }
```

Comment:
What do you think about:

Optional.ofNullable(getCorrelationId(routingContext)).ifPresent(id -> jsonResponse.put("correlationId", id ))